designcompose/src/main/java/com/android/designcompose/squoosh/SquooshAnimate.kt

```diff
@@ -74,11 +98,52 @@ internal class SquooshAnimatedLayout(
             return
         }
 
+        val fromWidth =
+            if (transformedChanged) from.view.style.fixedWidth(density) else fromLayout.width
+        val fromHeight =
+            if (transformedChanged) from.view.style.fixedHeight(density) else fromLayout.height
+        val toWidth = if (transformedChanged) to.view.style.fixedWidth(density) else toLayout.width
+        val toHeight =
+            if (transformedChanged) to.view.style.fixedHeight(density) else toLayout.height
+
+        if (transformedChanged) {
+            // Interpolate the decomposed matrix values, then construct a new matrix
+            /*
+            val target = DecomposedMatrix2D()
+            target.translateX = toDecomposed.translateX * value + fromDecomposed.translateX * iv
+            target.translateY = toDecomposed.translateY * value + fromDecomposed.translateY * iv
+            target.angle = toDecomposed.angle * value + fromDecomposed.angle * iv
+            target.scaleX = toDecomposed.scaleX * value + fromDecomposed.scaleX * iv
+            target.scaleY = toDecomposed.scaleY * value + fromDecomposed.scaleY * iv
+            target.skewXY = toDecomposed.skewXY * value + fromDecomposed.skewXY * iv
+            target.skewYX = toDecomposed.skewYX * value + fromDecomposed.skewYX * iv
+            this.target.style = this.target.style.withNodeStyle { s -> s.transform = Optional.of(target.toMatrix().toFloatList()) }
+            println("### From ${fromTransform}")
+            println("###   Angle ${fromDecomposed.angle} scaleX ${fromDecomposed.scaleX} scaleY ${fromDecomposed.scaleY}")
+            println("### To ${toTransform}")
+            println("###   Angle ${toDecomposed.angle} scaleX ${toDecomposed.scaleX} scaleY ${toDecomposed.scaleY}")
+            println("### Skew ${this.target.style.node_style.transform.get()}")
+
+             */
```

Comment:
Haven't been able to get the matrix reconstruction with skew to work.

Reply:
It looks like WebKit doesn't... but I'm not sure if the `decompose2`/`recompose2` methods handle skew or not. I think since the skew terms are included in `m11`, `m12`, `m21`, `m22` that skew is incorporated, but this blending function doesn't include those terms. 

* [recompose2](https://github.com/WebKit/WebKit/blob/main/Source/WebCore/platform/graphics/transforms/TransformationMatrix.cpp#L1804)
* [blend2](https://github.com/WebKit/WebKit/blob/main/Source/WebCore/platform/graphics/transforms/TransformationMatrix.cpp#L1804)

Reply:
Oh thanks, this is the code I was looking for earlier but couldn't find. Let me try it.

Reply:
Updated to use decomposition, works now 👍